html/cross-origin-embedder-policy/sandbox.https.html

```diff
@@ -0,0 +1,43 @@
+<!doctype html>
+<meta charset=utf-8>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/get-host-info.sub.js"></script>
+<div id=log></div>
+<script>
+async_test(t => {
+  window.addEventListener("message", t.step_func_done(({ data }) => {
+    assert_equals(data.origin, "null");
+    assert_true(data.sameOriginWithoutCORP, "Request to same-origin resource without CORP did not fail");
+    assert_true(data.sameOriginWithSameOriginCORP, "Request to same-origin resource with same-origin CORP did not fail");
+    assert_true(data.sameOriginWithCrossOriginCORP, "Request to same-origin resource with cross-origin CORP did not succeed");
+    assert_true(data.crossOriginWithCrossOriginCORP, "Request to cross-origin resource with cross-origin CORP did not succeed");
+  }));
+
+  const origins = get_host_info();
+  const frame = document.createElement("iframe");
+  const nothingCrossOriginCORP = new URL("resources/nothing-cross-origin-corp.txt", window.location).pathname;
+  const nothingSameOriginCORP = new URL("resources/nothing-same-origin-corp.txt", window.location).pathname;
+  frame.sandbox = "allow-scripts";
+  frame.srcdoc = `<script>
+let i = 0;
+const data = { sameOriginWithoutCORP: false,
+               sameOriginWithSameOriginCORP: false,
+               sameOriginWithCrossOriginCORP: false,
+               crossOriginWithCrossOriginCORP: false,
+               origin: self.origin };
+function report(token, success) {
+  i++;
+  data[token] = success;
+  if (i == 4) {
+    parent.postMessage(data, "*");
+  }
+}
+fetch("/common/blank.html", { mode: "no-cors" }).then(() => report("sameOriginWithoutCORP", false), () => report("sameOriginWithoutCORP", true));
```

Comment:
I forgot if I left a similar comment. Ignore this in that case. In any case this comment is optional.

I prefer Promise.all to manually counting things. Can we write something like:

function record(promise, token, expectation) {
  return promise.then(() => data[token] = expectation, () => data[token] = !expectation);
}

Promise.all([record(fetch(...), "sameOriginWithoutCORP", false),
                    record(fetch(...), "sameOriginWithSameOriginCORP", false),
                    record(fetch(...), "sameOriginWithCrossOriginCORP", true),
                    record(fetch(...), "crossOriginWithCrossOriginCORP", true)]).then(() => parent.postMessage(data, "*"));

?

Ditto for html/cross-origin-embedder-policy/resources/script-factory.js.